swift/looker/Tests/lookerTests/transportTests.swift

```diff
@@ -133,12 +133,86 @@ class transportTests: XCTestCase {
         XCTAssertEqual(user.email, "zz@foo.bar")
     }
 
+//    func dictToJson(dict: StringDictionary<Variant?>) -> String {
+//        var result = ""
+//        dict.flatMap({(arg: (key: String, value: Variant?)) -> String in let (key, value) = arg; return {
+//            let v = value?.toJson() ?? "null"
+//            return "\"\(key)\":\(v)" }
+//        })
+//        return result
+//    }
+
+    let visJson = """
+{
+"bool":true,
+"int":1,
+"dub":2.3,
+"str":"Simple string",
+"date":"2018-03-15T13:16:34.692-07:00",
+"nada":null,
+"dict": {"A":4, "B": 2, "C": true},
+"ratnest": [ { "one": 1, "two": "two" }, "three", {"four":4} ]
+}
+"""
+
+    // Relevant SO https://stackoverflow.com/questions/46279992/any-when-decoding-json-with-codable
+    // Using AnyCodable from https://github.com/Flight-School/AnyCodable
+    func testDictFromJson() {
+        var vis_config: StringDictionary<AnyCodable>? = try! deserialize(visJson)
+//        let vis_config: StringDictionary<Variant?>? = try! deserialize(visJson)
+        var data = try! serialize(vis_config)
+        var json = String(decoding: data, as: UTF8.self)
+        XCTAssertNotNil(vis_config)
+        XCTAssertNotNil(data)
+        XCTAssertNotNil(json)
+        let bool = vis_config["bool"] as! Bool
+        XCTAssertTrue(bool)
+        let int = vis_config["int"] as! Int64
+        XCTAssertEqual(int, 1)
+        let nada = vis_config["nada"]
+
+        if "\(nada!)" == "nil" {
+            // nada is nil as expected
+        } else {
+            XCTAssertTrue(false, "nada should be nil, not '\(nada!)'")
+        }
+        let dub = vis_config["dub"] as! Double
+        XCTAssertEqual(dub, 2.3)
```

Comment:
Floating point comparison :clench:

Maybe use this?
https://developer.apple.com/documentation/xctest/xctassertequalwithaccuracy

Reply:
Oh yeah, thanks for the reminder. I meant to do that. Will update. The version you link here is obsolete but there is an `accuracy` tag for `XCTAssertEqual`